src/compute/src/render/mod.rs

```diff
@@ -433,28 +437,28 @@ where
             );
 
             let token = traces.to_drop().clone();
-            let (mut ok_arranged, ok_button) = traces.oks_mut().import_frontier_core(
-                &self.scope.parent,
+            // Import the specialized trace handle as a specialized arrangement import.
+            // Note that we incorporate optional logging setup as part of this process,
+            // since a specialized arrangement import require us to enter a scope, but
+            // we can only enter after logging is set up. We attach logging here instead
+            // of implementing it in the server.rs worker loop since we want to catch the
+            // wall-clock time of the frontier advancement for each dataflow as early as
+            // possible.
```

Comment:
> we can only enter after logging is set up

Why is that? Can't we install the logging operator on the scoped stream too?

Reply:
AFAICT, the issue here is that we'd like to see the frontier advancements for the index before it's in the dataflow scope. This is because the dataflow scope could either be iterative or not. For the former, we'd see `Pointstamp` as the timestamp type, which our logging of import frontiers is not prepared to process.